service/resource/deployment/update.go

```diff
@@ -65,10 +65,6 @@ func (r *Resource) NewUpdatePatch(ctx context.Context, obj, currentState, desire
 }
 
 func (r *Resource) newUpdateChange(ctx context.Context, obj, currentState, desiredState interface{}) (interface{}, error) {
-	customObject, err := key.ToCustomObject(obj)
-	if err != nil {
-		return nil, microerror.Mask(err)
-	}
```

Comment:
There is now some code removed because in some places we only needed the parsed custom object for logging. 🎉